core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerServiceSpec.js

```diff
@@ -0,0 +1,344 @@
+                '<p>all others can be ruled out by definition of slope.</p>'
+              ],
+              param_changes: []
+            },
+            answer_groups: [
+              {
+                rule_specs: [
+                  {
+                    rule_type: 'Equals',
+                    inputs: {
+                      x: 0
+                    }
+                  }
+                ],
+                outcome: {
+                  dest: 'Question 1',
+                  feedback: [
+                    '<p>y/x is the measure of the slope.</p>'
+                  ],
+                  param_changes: []
+                }
+              }
+            ],
+            confirmed_unclassified_answers: [],
+            customization_args: {
+              choices: {
+                value: [
+                  '<p>y/x</p>',
+                  '<p>-y/x</p>',
+                  '<p>x/y</p>',
+                  '<p>-x/y</p>'
+                ]
+              }
+            },
+            id: 'MultipleChoiceInput'
+          },
+          content: [
+            {
+              type: 'text',
+              value: '<p><b>Linear algebra</b> is a subtopic of maths.</p>'
+            }
+          ],
+          unresolved_answers: {}
+        }
+      }
+    };
+
+    explorationStatesService.init(data.states);
+    explorationTitleService.init(data.title);
+    explorationInitStateNameService.init(data.init_state_name);
+
+    questions = statesToQuestionsService.getQuestions();
+    questionList = questionListObjectFactory.create(questions);
+
+    spyOn(validatorsService, 'isValidEntityName').andReturn(true);
+    spyOn(mockExplorationData, 'autosaveChangeList').andReturn(true);
+  });
+
+  it('should initialize the local data variables', function() {
+    var expectedReturnedValue;
+    // Case:1 when there are two valid states  with no error in initialized
+    // data, initialized in beforeEach block and StatesToQuestionsService
+    // .getQuestions() will return non null value.
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(simpleEditorManagerService.getTitle()).toBe(data.title);
+    expect(simpleEditorManagerService.getIntroductionHtml())
+      .toBe(data.states.Introduction.content[0].value);
+    // Expected questions are genereated from test data.
+
+    var expectedSimpleEditorManagerQuestionList = questionList;
+    expect(simpleEditorManagerService.getQuestionList())
+      .toEqual(expectedSimpleEditorManagerQuestionList);
+    expect(expectedReturnedValue).toBeTruthy();
+
+    // Case:2 when StatesToQuestionsService.getQuestions() will return null,
+    // may be due to errors.
+    spyOn(statesToQuestionsService, 'getQuestions').andReturn(null);
+    expectedReturnedValue = simpleEditorManagerService.tryToInit();
+    expect(expectedReturnedValue).toBeFalsy();
+  });
+
+  it('should return data', function() {
+    // Expected initialized data inside simpleEditorManagerService
```

Comment:
Comment probably not needed; it's explained well by the variable name.

In general, use comments to explain why something occurs or for other stuff that's not clear by looking at the code. But in general it's better to pick good variable names and structure things explicitly so that the code is as self-explanatory as possible.